nengo_loihi/loihi_cx.py

```diff
@@ -686,24 +686,26 @@ def step(self):  # noqa: C901
 
         noise = self.noiseGen()
         q0[self.noiseTarget == 0] += noise[self.noiseTarget == 0]
+        _, o = overflow_signed(q0, bits=Q_BIT, out=q0, return_hits=True)
+        if np.any(o):
+            self.error("Overflow in q")
 
-        # self.U[:] = self.decayU_fn(self.U, self.decayU, a=12, b=1)
         self.u[:] = self.decayU_fn(self.u[:], q0)
-        u2 = self.u[:] + self.bias
+        _, o = overflow_signed(self.u, bits=U_BIT, out=self.u,
+                               return_hits=True)
+        if np.any(o):
+            self.error("Overflow in U")
+        u2 = self.u + self.bias
         u2[self.noiseTarget == 1] += noise[self.noiseTarget == 1]
-        if np.any(u2 > U_MAX):
-            self.error("Overflow in U (max was %d)" % u2.max())
-        if np.any(u2 < U_MIN):
-            self.error("Underflow in U (min was %d)" % u2.min())
-        u2 = np.clip(u2, a_min=U_MIN, a_max=U_MAX, out=u2)
+        _, o = overflow_signed(u2, bits=U_BIT, out=u2, return_hits=True)
+        if np.any(o):
+            self.error("Overflow in u2")
 
-        # self.V[:] = self.decayV_fn(v, self.decayV, a=12) + u2
         self.v[:] = self.decayV_fn(self.v, u2)
-        if np.any(self.v > V_MAX):
-            self.error("Overflow in V (max was %d)" % self.v.max())
-        if np.any(self.v < V_MIN):
-            self.error("Underflow in V (min was %d)" % self.v.min())
-        self.v = np.clip(self.v, a_min=V_MIN, a_max=V_MAX, out=self.v)
+        # _, o = overflow_signed(self.v, bits=V_BIT, out=self.v,
+        #                        return_hits=True)
+        # if np.any(o):
+        #     self.error("Overflow in V")
```

Comment:
Why is this part commented out?  Do we not need to check for overflow in V?  Or is this related to the comment in the test, that you couldn't find a way to get V to overflow?

Reply:
Exactly. I feel like it should overflow, and that this code should thus be in there, but I couldn't actually figure out a way to hit this and test the code, so I commented it out.

Reply:
I am pretty sure that I've gotten an overflow in V in the Nengo tests... I'll try to track it down and see if it can be distilled into a test.